### PR TITLE
services/horizon/expingest: Update DatabaseProcessor with trust lines

### PR DESCRIPTION
**Pull request by bartekn**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### Summary

This commit adds trust lines to ingestion pipeline.

### Goal and scope

In https://github.com/stellar/go/issues/1765, https://github.com/stellar/go/issues/1766 and https://github.com/stellar/go/issues/442 we need trust lines in Horizon DB to serve the requests. This commit is adding a code needed by all of the issues above.

### Summary of changes

* Adds a new migration file adding a new `trust_lines` table.
* Updates `db2/history` package with trust lines related interfaces and structs.
* Updates `DatabaseProcessor` to process trust lines in state and ledger pipelines.
* Upgrades `Masterminds/squirrel` to support sorting fields list when using `SetMap` (to be able to assert equality in tests). List of dependency changes (used [`golistcmp`](https://github.com/stellar/golistcmp)):
```
  0	github.com/Masterminds/squirrel	20f192218cf5	>	v1.1.0		https://github.com/Masterminds/squirrel/compare/20f192218cf5...v1.1.0	
  1	github.com/davecgh/go-spew	v1.1.0		>	v1.1.1		https://github.com/davecgh/go-spew/compare/v1.1.0...v1.1.1		
  2	github.com/lann/builder		c603884a2c1f	>	47ae307949d0	https://github.com/lann/builder/compare/c603884a2c1f...47ae307949d0	
  3	github.com/lann/ps				+	62de8c46ede0	https://github.com/lann/ps						
  4	github.com/mndrix/ps		33ddf69629c1	-			
```
For 3 and 4 the compare view is here: https://github.com/lann/ps/compare/33ddf69629c1...62de8c46ede0

### Known limitations & issues

* ~No tests for the new code.~
* ~State verifier not updated to check trust lines.~ Done in 3b74910.

However, if you want to start working on any of https://github.com/stellar/go/issues/1765, https://github.com/stellar/go/issues/1766 and https://github.com/stellar/go/issues/442 you can branch off from this PR branch.

### CODE REVIEW

**Comment by tamirms** [comment]
@bartekn to support the `/assets` endpoint it seems like we'll need a table equivalent to `asset_stats` from the old ingestion system. Will that be implemented in this PR?

**Comment by bartekn** [comment]
@tamirms no, this PR only adds trust lines (shared code needed in #1765, #1766 and #442). `asset_stats` table should be added in #1765).

**Comment by bartekn** [comment]
I think this PR is ready for review.

When it comes to tests, I'd like to continue discussion started here: https://github.com/stellar/go/pull/1699#discussion_r323455894. I created tests for `TrustLinesQ` methods that are not using postgres but mocked version of `db.Session`. I did this because the idea here is slightly different than what @tamirms described in his PR: it's not storing data in memory, it just checks if the queries and arguments are properly built before passing them to `Exec` function. Unfortunately it required small changes in `history.Q` to allow mocking. Let me know what you think, it's easy to revert the changes and write tests that use postgres.

**Comment by tamirms** [comment]
@bartekn the code looks good to me. Regarding the tests, I would prefer if services/horizon/internal/db2/history/trust_lines_test.go interacted with postgres. In the current tests, if the sql query is not correct (e.g. there is a typo in one of the columns, or a column is missing, or there is a violation of some constraint) we would be relying on code review to catch these errors. I think if the tests ran the queries against postgres we would have better chances of catching these types of errors even if they slipped through code review.

**Comment by bartekn** [comment]
@tamirms ok, tests are using DB now. PTAL!
> In the current tests, if the sql query is not correct (e.g. there is a typo in one of the columns, or a column is missing, or there is a violation of some constraint) we would be relying on code review to catch these errors.

Yes, but such things should be checked during integration tests run in CI. The holy grail, imho, is to have unit tests that are fast and doesn't require any 3rd party tool installed (including docker). But I agree that we should probably stick to the old method until we have a good integration tests in CI.

File: services/horizon/internal/db2/history/trust_lines_test.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+					Buying:  1,
+					Selling: 2,
+				},
+			},
+		},
+	}
+)
+
+func TestInsertTrustLine(t *testing.T) {
+	tt := test.Start(t).Scenario("base")
```
Comment (tamirms) [inline services/horizon/internal/db2/history/trust_lines_test.go]: actually you don't need to use scenarios, you can use this code:

```
	tt := test.Start(t)
	defer tt.Finish()
	test.ResetHorizonDB(t, tt.HorizonDB)
	q := &Q{tt.HorizonSession()}
```

**Pull request merged**